.github/workflows/tests.yml

```diff
@@ -0,0 +1,90 @@
+name: build and test
+
+on: [push, pull_request]
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.14', '1.15']
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # clone in the gopath
+          path: src/github.com/${{ github.repository }}
+      - uses: actions/setup-go@v2
+        with:
+          stable: false
+          go-version: ${{ matrix.go }}
+      - name: setup environment
+        run: |
+          echo "GOPATH=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "GO111MODULE=on" >> $GITHUB_ENV
+      - name: build coredhcp
+        run: |
+          set -exu
+          cd $GITHUB_WORKSPACE/src/github.com/${{ github.repository }}/cmds/coredhcp
+          go build
+  coredhcp-generator:
```

Comment:
This one is more of a lint than a test I think ? it's just checking that the 2 `main.go` don't drift from each other

Reply:
I guess it's border-line: we are building coredhcp-generator (which is a test per-se) and running it to build a source file, and then comparing it. I'm ok with moving it under `lint.yml` but it won't be a perfect fit there either

Reply:
fair enough let's leave it like this then